### ISSUE TITLE
Make table-format functionality available outside of tabulars

### ISSUE THREAD

**Issue by schtandard** [josephwright/siunitx#403]
Sometimes I want to horizontally align numbers in contexts other than a `tabular` environment (e.g. in the y-tick labels of a plot). Since the machinery to do this is already present in `siunitx` for tables, I expect making it available in other places wouldn't be too much of a hassle.

I'm imagining an option to `\SI` and similar macros that, instead of using the natural width of the printed number, uses the format set in `table-format` (or an equivalent option) to typeset the number in a box of predetermined width. This should happen in such a way that stacking these boxes below each other leads to the same alignment as an `S` column would.

Basically, I want to conveniently print single table cells (both in math and text mode).

**Comment by josephwright** [comment]
That's what happens with `\tablenum`, more-or-less. Could you provide a demo?

**Comment by josephwright** [comment]
As a demo
```latex
\documentclass{article}
\usepackage{siunitx}
\begin{document}

\sisetup{table-format = +3.3}
\fbox{\tablenum{12.345}}

\end{document}
```

**Comment by schtandard** [comment]
That's exactly what I was looking for, seems I missed it in the documentation.

Well, actually I was also imagining something like a `\tableSI` basically equivalent to
```latex
\NewDocumentCommand\tableSI{ O{} m m }{%
  \tablenum[#1]{#2}\l__siunitx_number_unit_product_tl\si{#3}%
}
```
However, I think that this is only useful in edge cases.
